common/src/main/java/earth/terrarium/hermes/utils/ElementParsingUtils.java

```diff
@@ -116,7 +117,48 @@ public static Alignment parseAlignment(Map<String, String> parameters, String ke
         return defaultValue;
     }
 
-    public static <A,B, C> @Nullable C parsePair(
+    public static Map<String, String> parseStyleAttribute(Map<String, String> parameters) {
+
+        // Allow for using 'style="..."' attribute values to compactly specify multiple attributes.
+        // These compact attributes may be in 'key:value' or boolean form.
+        // Booleans can be negated with a leading '!'; both forms may have short names or synonyms.
+        // Example: style="!obfuscated,b,em,color:lavender,bg:0 #4a38d0"
+
+        Map<String, String> result = new HashMap<>();
+        String[] specificers = parameters.get("style").split(",");
+        String specValue;
+        for (String specifier : specificers) {
+            String[] specParts = specifier.split(":",2);
+            if (specParts.length == 2) {
+                // key:value form
+                specifier = specParts[0];
+                specValue = specParts[1];
+            } else {
+                // boolean form
+                if (specifier.startsWith("!")) {
```

Comment:
I think for such a style the boolean negate should not be allowed it should stay close to how the style attribute in html works.

Reply:
TBH, I don't see it that way for this one.

- Style negation immediately struck me as a _fantastic_ idea when I first saw the negation ChildTextTags you came up with.
- Hermes is far implementing standard HTML and CSS (and it should be) and has already departed from being a subset of HTML/CSS in order to serve it's use, through special tags `<component>`, `<detail>`, `<hint>`, `<carousel>`, `<entity>`, `<column(s)>` and `<item>`, tags, some Minecraft specific styling conventions, a much simpler layout model, etc.
- If HTML is the reference point, this would a backwards compatible addition.

Reply:
okay yeah makes sense I think its fine then.